pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,829 @@
+        return sane_dictionary
+
+    ################
+    # System Setup #
+    ################
+
+    def default_setup(self):
+        """ Setup the oscilloscope for remote operation. """
+        self.write("CHDR OFF")
+
+    def ch(self, channel_number):
+        if channel_number == 1 or channel_number == "C1":
+            return self.ch1
+        elif channel_number == 2 or channel_number == "C2":
+            return self.ch2
+        elif channel_number == 3 or channel_number == "C3":
+            return self.ch3
+        elif channel_number == 4 or channel_number == "C4":
+            return self.ch4
+        else:
+            raise ValueError("Invalid channel number. Must be 1, 2, 3, 4.")
+
+    def autoscale(self):
+        """ Autoscale displayed channels. """
+        self.write("ASET")
+
+    ##################
+    # Timebase Setup #
+    ##################
+
+    timebase_offset = Instrument.control(
+        "TRIG_DELAY?", "TRDL %eS",
+        """ A float parameter that sets the time interval in seconds between the trigger
+        event and the reference position (at center of screen by default)."""
+    )
+
+    timebase_scale = Instrument.control(
+        "TDIV?", "TDIV %eS",
+        """ A float parameter that sets the horizontal scale (units per division) in seconds (S), milliseconds (MS), 
+        microseconds (US) or nanoseconds (NS) for the main window.""",
+        validator=strict_range,
+        values=[1e-9, 100]
+    )
+
+    timebase_hor_magnify = Instrument.control(
+        "HMAG?", "HMAG %eS",
+        """ A string parameter that sets the zoomed (delayed) window horizontal scale (seconds/div). The main sweep 
+        scale determines the range for this command. """,
+        validator=strict_range,
+        values=[1e-9, 20e-3]
+    )
+
+    timebase_hor_position = Instrument.control(
+        "HPOS?", "HPOS %eS",
+        """ A string parameter that sets the horizontal position in the zoomed (delayed) view of the main sweep. The 
+        main sweep range and the main sweep horizontal position determine the range for this command. The value for 
+        this command must keep the zoomed view window within the main sweep range.""",
+    )
+
+    @property
+    def timebase(self):
+        """ Read timebase setup as a dict containing the following keys:
+            - "seconds_div": horizontal scale in seconds/div (float)
+            - "delay": interval in seconds between the trigger event and the reference position (float)
+            - "hor_magnify": horizontal scale in the zoomed window in seconds/div (float)
+            - "hor_position": horizontal position in the zoomed window in seconds (float)"""
+        tb_setup = {
+            "seconds_div": self.timebase_scale,
+            "delay": self.timebase_offset,
+            "hor_magnify": self.timebase_hor_magnify,
+            "hor_position": self.timebase_hor_position
+        }
+        return {k: float(v) for k, v in tb_setup.items()}
```

Comment:
again, why the additional cast here? What problems did you encounter that made this necessary? Our properties should return float by default (if parseable)

Reply:
Fixed in comment above: https://github.com/pymeasure/pymeasure/pull/697#discussion_r985674026